### ISSUE TITLE
[Bug Report] Audio Output Not Working in Freeshow App

### ISSUE THREAD

**Issue by jquyatt** [ChurchApps/FreeShow#846]
**Describe the bug**
The Freeshow app refuses to play audio from video slides or audio files through the system output. Despite seeing the audio meters moving beside the preview, no sound is coming through the speakers. All unmute buttons were tested but did not resolve the issue. Other audio sources (QuickTime, Spotify) are working fine, confirming that the issue is isolated to the Freeshow app.

**Reproducing**

1.	Open the Freeshow app.
2.	Load a video slide or audio file.
3.	Attempt to play the audio.
4.	Observe that no sound is output, despite the audio meter moving next to the preview.

Expected Behavior:
Audio should play through the system output when the video slide or audio file is played within the Freeshow app.

Actual Behavior:
No audio is heard through the system output, even though the audio meter indicates activity.

System Information:

•Device: MacBook Pro (13-inch, M1, 2020)
•Chip: Apple M1
•Memory: 8 GB
•macOS Version: 14.6.1 (23G93)
•Freeshow Version: 1.2.7

**Additional context**
Troubleshooting Steps Taken:

•Verified the system output is working by testing with QuickTime and Spotify.
•Attempted all available unmute options within the Freeshow app without success.

Additional Notes:

•No changes in system audio settings or connected devices during this period.

**Comment by vassbo** [comment]
Thanks.

Ensure the volume is set to 100.
![bilde](https://github.com/user-attachments/assets/5731beef-0699-4b11-8c82-20fb3e250cc3)

Also you can try tweaking the audio output in "Audio settings", leaving at blank should use the system default.
![bilde](https://github.com/user-attachments/assets/d446e626-3fd5-4bf3-9a4d-46e8f1686fe1)

Lastly try updating to the latest pre-release: https://github.com/ChurchApps/FreeShow/releases/tag/v1.2.8-beta.1

**Comment by jquyatt** [comment]
I 100% had the Audio fader hidden and couldn't see that it was set to zero. Thanks for the help!

**Comment by NikolasGregor** [comment]
i try to connect a source via audio>microphones>sound source which can be a sound console and the sound is totally corrupted and has been heard multiple times, in other applications like obs studio there is no problem

1.3.0v

**Comment by vassbo** [comment]
@NikolasGregor Please test this in Google Chrome and let me know if you have the same sound issues or not: https://mictests.com/

**Comment by vassbo** [comment]
You have the "Gain" at 0, right? Gaining can cause distortion!
![bilde](https://github.com/user-attachments/assets/461ee253-6703-41a9-bbd3-e4cce2585bdc)

**Comment by NikolasGregor** [comment]
Yes gain is 0. the problem is like filter noise suppression 100%

**Comment by vassbo** [comment]
They have pretty much the same playback logic, but I think I found the issue..
The microphones are playing double, in both an audio stream and a audio object.

**Comment by thebiblelover7** [comment]
@vassbo I had this same issue trying to set FreeShow up today. The audio sounds like noise suppression is set to 100%. Also, the volume fader doesn't seem to do anything on my tests... Hope this can get fixed soon :)

**Comment by vassbo** [comment]
@NikolasGregor Okay!
@thebiblelover7 You said you also had a similar issue, would you mind checking it out if you still have mic audio issues?

**Comment by thebiblelover7** [comment]
@vassbo I'm still having issues with it. The audio still sounds like it is being compressed and noise reduced a lot. It isn't usable at all...

**Comment by thebiblelover7** [comment]
@Dcanadian I'm using a usb-c capture card that also brings audio with it. It works perfectly in ProPresenter, but not in FreeShow.

**Comment by thebiblelover7** [comment]
Yes, I could try to do that tomorrow

**Comment by thebiblelover7** [comment]
It's been a while, and I still have the same issue on the latest beta (after the audio refactoring). @vassbo Could you help me debug this sometime? I was having issues debugging it...

**Comment by vassbo** [comment]
I can try, where do you get stuck in the debugging?

**Comment by lehik** [comment]
https://github.com/ChurchApps/FreeShow/pull/1557

**Comment by thebiblelover7** [comment]
Thanks @lehik ! That solved the problem completely. We were able to use it for our church service yesterday without any issues!

**Comment by levyian** [comment]
Hi @lehik, I'm hoping you could help me with the same issue. I'm on version 1.4.7 and I can't get audio out when playing video files. Audio works when using the built in music player (mp3), but not video slides nor adding the media directly to the slides area.

The video files do play outside freeshow.

(Edit) I'm on Windows 11 and my volume is 100% and no custom audio output.

**Comment by vassbo** [comment]
@levyian Maybe you can provide a screen recording of the steps you are taking so that we might give a clue to what is happening?

**Comment by levyian** [comment]
Hey @vassbo, of course. I'll do that this evening when I get back to my pc. Thanks.

-------- Original Message --------

**Comment by levyian** [comment]
@vassbo, here is the screen recording: https://drive.proton.me/urls/S5BC6QXXF4#sDhDR4tdydfb.

0:0 I tried playing the video directly, no sound
0:15 I add it as a slide, but not sound
0:33 I played an mp3 file through the audio section and got sound

**Comment by vassbo** [comment]
Thanks, that's strange. You seem to be doing everything correctly. Can you go to settings>Outputs and turn off all outputs except for one, then restart the app, then click the "Present" button in the top right corner, then try playing a video to see if it's an output that's the issue?

**Comment by marcosmolina7** [comment]
> Thanks, that's strange. You seem to be doing everything correctly. Can you go to settings>Outputs and turn off all outputs except for one, then restart the app, then click the "Present" button in the top right corner, then try playing a video to see if it's an output that's the issue?

Hello everyone, maybe this information will help, to me it happens to me that does not give the audio doing the following:

1. I have 3 outputs, the main and two NDI, in the NDI outputs does not give the audio, unless I mark in the main output that if it carries audio. Then, even if I check the other two NDI outputs to send audio, it does not do it if the main output does not have this option activated.

2. All this arose because I do not need the NDI activated in the main output, but if in the other two NDI outputs, which in fact, these two NDI outputs I have with the option that they are invisible, once I deactivate the NDI of the main output, it stops giving the audio or if I remove the audio from the main output, then it affects the audio of the other two NDI outputs.

3. So according to me the outputs are not independent in that aspect. And another thing, is there a way that the transparency option can be enabled no matter if NDI is enabled or not? As I mentioned before, with me it is not necessary to have in that main output the NDI on but in the other 2, but in the main one if I need it to be transparent but that option is only having the NDI on, so for me, resources are being wasted with a NDI on that is not necessary to use and that I just need to have the option that the output is transparent.

Thanks for this great work with Freeshow, your support and follow up is appreciated. Blessings!

**Comment by vassbo** [comment]
@marcosmolina7 The audio will only play in the first output if playing the same video! Also why would you need a transparent output if you don't use NDI?

**Comment by marcosmolina7** [comment]
> @marcosmolina7 The audio will only play in the first output if playing the same video! Also why would you need a transparent output if you don't use NDI?

I understand, but if I use the same video or resource for the 3 outputs, but if to the first output (the one that says main), I remove the audio in the option of NDI with audio, the other two independent outputs of NDI no longer send the audio even if the sending is activated in these NDI outputs. And I use transparency, since the bible is managed with another program, and I need to put the Freeshow vignettes (layers), so that only that layer comes out.
